### ISSUE TITLE
Feature request: pass `-S/--strains` to `ncbi-genome-download`

### ISSUE THREAD

**Issue by oxygen311** [gem-pasteur/PanACoTA#17]
Hello, Amandine!

In one of mine specific task, I need to download specific strains from NCBI: every strain from one species + one outgroup strain. I would like to still use `PanACoTA` pipeline for analysis but didn't found a way to do it easily.

I think, adding the ability to pass `-S/--strains` to `ncbi-genome-download` will solve the problem. Could you please add this feature to PanACoTA?

**Comment by asetGem** [comment]
Hello,

I cannot check for now, but I think that what you may want to use is the `-t ncbi_taxid` option (of `prepare` module).
Option `-T` is for all genomes of a species, while `-t` is for a subspecies or specific strain.
You can also use both, and ask for several specific strains by giving all strain IDs separated by a comma (`-t id1,id2,id3`)

But I am not sure it corresponds to what you want. Let me know!

**Comment by oxygen311** [comment]
As I can read in `ncbi-genome-download` docs, `-t` is only for NCBI taxonomy IDs:

```
-T SPECIES_TAXIDS, --species-taxids SPECIES_TAXIDS
                        Only download sequences of the provided species NCBI taxonomy IDs. A comma-separated list of species taxids is also possible. For example: "52342,12325". (default: [])
-t TAXIDS, --taxids TAXIDS
                        Only download sequences of the provided NCBI taxonomy IDs. A comma-separated list of taxids is also possible. For example: "9606,9685". (default: [])
```

But unfortunately, most strains within some species have the same taxid which is equal to species taxid. Here is an example of some strains of _Burkholderia mallei_:
<img width="398" alt="Screenshot 2021-07-07 at 18 28 04" src="https://user-images.githubusercontent.com/16215034/124787314-130e9580-df51-11eb-805d-925b4a8a1279.png">

**Comment by asetGem** [comment]
Do you mean that you do not want to download all strains with taxid ==13373 but only some of them (like Turkey1, Ivan for example)?

**Comment by oxygen311** [comment]
> Do you mean that you do not want to download all strains with taxid ==13373 but only some of them (like Turkey1, Ivan for example)?

Yes :)

More specifically, I want only one strain of _Burkholderia pseudomallei_ for the outgroup (strain=VB29710, taxid=28450, species_taxid=28450) and all strains of _Burkholderia mallei_ (taxid=13373) for analysis. Setting outgroup can resolve root location problem in phylogenetic tree and some other problems too.

**Comment by asetGem** [comment]
Yes, I understand what you mean. 
Do you know if there is an option in `ncbi-genome-dowload` where you can give a strain name?
I do not have access to any computer, so I cannot check anything.

**Comment by oxygen311** [comment]
> Yes, I understand what you mean.
> Do you know if there is an option in `ncbi-genome-dowload` where you can give a strain name?
> I do not have access to any computer, so I cannot check a  -S STRAINS, 

Yes, I found, that desired protocol can be achieved with `--strains` flag of `ncbi-genome-dowload`:
```--strains/-S STRAINS

Only download sequences of the given strain(s). A comma-separated list of strain names is possible, as well as a path to a filename containing one name per line.```

**Comment by asetGem** [comment]
Ok, I can add this option to panacota `prepare` module, but I will not be able to do it before the end of the month.
If (and I guess yes) you need it before, you can download the VB29710 sequence with ngd, and add it to all 133373 strains downloaded by panacota.
Then, you will be able to use (annotate, pan...) all strains as if they were downloaded together.

**Comment by oxygen311** [comment]
> Ok, I can add this option to panacota `prepare` module, but I will not be able to do it before the end of the month.
> If (and I guess yes) you need it before, you can download the VB29710 sequence with ngd, and add it to all 133373 strains downloaded by panacota.
> Then, you will be able to use (annotate, pan...) all strains as if they were downloaded together.

Ok, thank you. Then I'll try to download and add this strain with `ngd` for now.

**Comment by oxygen311** [comment]
Hello, Amandine!

I've decided to add this feature by myself and created pull request with all info about this improvement :)
I think you can use it as a codebase if you will add this feature to PanACoTA.

**Comment by asetGem** [comment]
Hello!

Thank you for the PR #18. I have now merged it with the main code, in version 1.3.1-dev1.
I also added some tests and fixed compatibility problems.

As you suggested, I checked that it works fine when giving a file instead of a list of strains.

However, I am not sure to understand your first point (in the PR #18  comment). Do you mean that from prepare module it does not have the same behaviour as in the all module?
